src/serial.rs

```diff
@@ -122,74 +88,259 @@ pub enum Error {
 
 pub trait Pins<USART> {
     const REMAP: u8;
```

Comment:
AFAIK remap does nothing, perhaps we should remove it.

Reply:
I thought about it, while implementing this, but i decided to leave it afterall.

I'd be happy to just remove it.. As far as i know the AFIO system completely replaced pin remapping on all stm32 cortex-m series, but cortex-m1?